common/src/main/java/io/pravega/common/util/ByteArraySegment.java

```diff
@@ -171,44 +172,28 @@ public void copyTo(OutputStream stream) throws IOException {
     }
 
     @Override
-    @SneakyThrows(IOException.class)
     public boolean equals(BufferView other) {
         if (this.length != other.getLength()) {
             return false;
         } else if (other instanceof ArrayView) {
-            return equals((ArrayView) other);
+            // ByteBuffer-optimized equality check.
+            return this.asByteBuffer().equals(((ArrayView) other).asByteBuffer());
```

Comment:
I believe this method does the same thing without the wrapping:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Arrays.html#equals(byte%5B%5D,int,int,byte%5B%5D,int,int)

Reply:
Good point. Fixed.

Reply:
That method is not available in JDK8. This class is part of `common` so it must compile with JDK8 APIs. 

However, instances of this class may be used in JRE11 environments (such as Segment store) where the updated `ByteBuffer.equals` will be used.